drizzlepac/hlautils/testutils.py

```diff
@@ -102,7 +110,8 @@ def compare_wcs_alignment(dataset):
         alignment[wcs] = extract_results(results)
 
     # Restore user environment to original state
-    os.environ['ASTROMETRY_STEP_CONTROL'] = control
+    if control:
+        os.environ['ASTROMETRY_STEP_CONTROL'] = control
```

Comment:
In my opinion, there is a serious issue with this change/restoring of the environment: if a crash occurs, the environment will not be restored to the previous state. It think this should be either addressed somehow or documented/explained as to why this should be OK.

Reply:
This was fixed by wrapping the code in a try...except...finally block which still preserves all traceback information and insures that no matter what happens, if the environment was changed it will be restored. :-)

Reply:
except the except block is not necessary :-)